gluten-core/src/main/scala/org/apache/gluten/backendsapi/IteratorApi.scala

```diff
@@ -91,4 +91,6 @@ trait IteratorApi {
       numOutputRows: SQLMetric,
       numOutputBatches: SQLMetric,
       scanTime: SQLMetric): RDD[ColumnarBatch]
+
+  def toLocalFilesNodeByteArray(p: GlutenRawPartition): Array[Array[Byte]]
```

Comment:
could we add a new SplitInfo object file and move this method into it with `toSplitInfoByteArray`? then other backends could use it more easily, and avoid add this method in `IteratorApi` which seems unrelated.